### DISCUSSION TITLE
PyInstaller.depend.bindepend' has no attribute 'selectImports

### DISCUSSION THREAD

**Discussion by Patricklevyy** [kivy/kivy-github-discussions#1] [Q&A]
Hello,
I'm trying to convert my .py file to .exe with py installer. I have a simple code that uses kivy. But I have an error that I can't fix. When I launch pyinstaller I get after a few moments:
`AttributeError: module 'PyInstaller.depend.bindepend' has no attribute 'selectImports'`
I asked the issue on pyinstaller discussion but the autor said : 

PyInstaller.depend.bindepend.selectImports was refactored into low-level PyInstaller.depend.bindepend.get_imports and high-level PyInstaller.depend.bindepend.binary_dependency_analysis in v.6.0 (https://github.com/pyinstaller/pyinstaller/pull/7784).
So if kivy hook is using selectImports, it should be fixed accordingly. As the hook in question is maintained as part of kivy, you should report the issue to them. (As a side note, I am not sure why kivy hook would need to perform binary dependency analysis itself - it should just return the top-level binaries and let PyInstaller handle the dependencies).

**Comment by ElliotGarbus** [comment]
What version of pyinstaller are you running?

**Comment by Heribert17** [comment]
Hello,
I get the same error. I'm using
Python3.12
pyinstaller version 6.4.0
I tested under Linux Zorin and Windows 10
There is no downgrade to pyinstaller 5.9 possible because the 5.9 version doesn’t support Python 3.12

**Reply by Patricklevyy** [reply]
Hello, have you tried downgrading to pyinstaller 5.12.0?
That's what I did to make it work and I was on windows 10 too with python 3.11.7.

**Reply by Heribert17** [reply]
Hello Patricklevyy,
pyinstaller 5.12. works with Python 3.11 but I'm using Python 3.12.2 and pyinstaller 5.12.0 doesn't work with Python 3.12. I need the pyinstaller 6.x version.
What's strange is that pyinstaller 6.4 and Python 3.12 and Kivy 2.3 work on Windows 10 
but not on Linux. It seems as if the pyinstaller hooks in Kivy where only changed for Windows.

**Comment by qlrd** [comment]
@Patricklevyy , I had same issue. Found a answer (not remember where 🤕 ) that simple patch it.

I used scripts to automatically patch and use pyinstaller with kivy:

- [bash script](https://github.com/qlrd/krux-installer/blob/kivy/patch-pyinstaller-kivy-hook.sh) 
- [powershell script](https://github.com/qlrd/krux-installer/blob/kivy/patch-pyinstaller-kivy-hook.ps1)

To me, it resolves.

**Comment by ATchibo** [comment]
I also have a python app using Kivy and ran into the same issue. However, when I used python 3.12.3 with pyinstaller 5.13.0 it worked for me.
